main.go

```diff
@@ -360,7 +361,23 @@ func getTracer(opts Options, out output) (opentracing.Tracer, io.Closer) {
 		closer io.Closer
 	)
 	if opts.TOpts.Jaeger && !opts.TOpts.NoJaeger {
-		tracer, closer = jaeger.NewTracer(opts.TOpts.CallerName, jaeger.NewConstSampler(true), jaeger.NewNullReporter())
+		var err error
+		tracer, closer, err = jaeger_config.Configuration{
+			ServiceName: opts.TOpts.CallerName,
+			Throttler: &jaeger_config.ThrottlerConfig{
+				SynchronousInitialization: true,
```

Comment:
can you explain this a little -- i'm assuming it means we try to get the throttling information synchronously? if that fails (e.g., on machines without jaeger), what is the impact?

Reply:
Synchronous initialization is required for scripting tools such as yab that will likely exist in a short-lived process. For a service, we can periodically fetch the credit information (i.e. every 10 seconds). That is not feasible in a short script, so we set this flag to indicate we need to request credits from the Jaeger agent synchronously if there are no credits available for a given operation (credits represent number of debug spans we can emit without violating the throttling API). On machines without Jaeger, this would fail the request and refuse to emit debug spans, which is fine for those machines without agents. I would have thought the initial query could take a while to determine that the agent is not running and return an error, but local tests indicate no significant lag.

Reply:
@isaachier add this justification as a code comment.